misk/src/main/kotlin/misk/hibernate/HibernateHealthCheck.kt

```diff
@@ -0,0 +1,56 @@
+package misk.hibernate
+
+import misk.healthchecks.HealthCheck
+import misk.healthchecks.HealthStatus
+import misk.jdbc.DataSourceConfig
+import org.hibernate.PessimisticLockException
+import org.hibernate.QueryTimeoutException
+import org.hibernate.SessionFactory
+import java.sql.Timestamp
+import java.time.Clock
+import java.time.Duration
+import javax.inject.Singleton
+import javax.persistence.LockTimeoutException
+import javax.persistence.NoResultException
+import javax.persistence.NonUniqueResultException
+import javax.persistence.PersistenceException
+import javax.transaction.TransactionRequiredException
+
+/**
+ * HealthCheck to confirm database connectivity and defend against clock skew.
+ */
+class HibernateHealthCheck(
```

Comment:
test?

Reply:
yep. Thoughts on how to do this? I'm thinking about how to do the clock skew since we use a FakeClock for tests but the time returned is from the database (if that's the test MySQL it will be the hosts time).

Reply:
you don't have to test the clock skew if it's difficult or flaky. but it seems like you could set the FakeClock to the real current time and it would be close enough.

Reply:
yep!

Reply:
Added tests.